File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter08/ConsoleListener/CarListenerClient/ClientMain.cs
using System;
using SimpleCarLibrary;

namespace CarListenerClient
{
   class CarMain
   {
      static void Main(string[] args)
      {
         
         // Activate the well known object
         CarService carSvc = (CarService) Activator.GetObject(
            typeof(SimpleCarLibrary.CarService),
            "tcp://localhost:13101/CarService"
         );
                          
         // Invoke a method
         do 
         {
            Console.WriteLine(carSvc.GetInfo());
         }while(Console.ReadLine() != "q");
      }
   }
}

File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter08/ConsoleListener/CarListenerClient/ClientMain.vb
Imports SimpleCarLibrary

Module CarMain

    Sub Main()
      'Activate the well-known object
      Dim remote As Object 
      remote = Activator.GetObject( _
                  GetType(CarService), _
                  "tcp://localhost:13101/CarService")
                  
      Dim carSvc As CarService = CType(remote, CarService)

      'Invoke a method
      Console.WriteLine(carSvc.GetInfo())
    End Sub
End Module
